src/parser/type_checking.c

```diff
@@ -11815,6 +11810,12 @@ pt_common_type_op (PT_TYPE_ENUM t1, PT_OP_TYPE op, PT_TYPE_ENUM t2)
       result_type = PT_TYPE_INTEGER;
     }
 
+  if (pt_is_comp_op (op) && (PT_IS_NUMERIC_TYPE (t1) && t2 == PT_TYPE_JSON)
+      || (t1 == PT_TYPE_JSON && PT_IS_NUMERIC_TYPE (t2)))
```

Comment:
It would be nice to clarify the condition:
It is (cond a && cond b || cond c)

You might want (cond a && (cond b || cond c))

Reply:
done